pkg/oci/containerize.go

```diff
@@ -150,27 +152,28 @@ func newDataTarball(source, target string, ignored []string, verbose bool) error
 			}
 		}
 
+		// Check for invalid links (absolute, outside of function root, etc)
+		if err := validateLink(root, path, info); err != nil {
+			return err
+		}
+
 		header, err := tar.FileInfoHeader(info, info.Name())
```

Comment:
@lkingland  Shouldn't be the second argument of `tar.FileInfoHeader()` link target?
Again this would have been caught if testing was done at least at level of `newDataTarball()` or better yet [builder.Build()](https://github.com/knative/func/blob/24fe6d36fa07787df3962c40bb299a1aab8f441c/pkg/oci/builder_test.go#L30) level, not at level of some measly auxilary function.

Reply:
Sure; please see the latest addition.   As I mentioned above in our first set of comments, I agree that a high-level test should be added.  Now, I do think there also needs to be a case added which specifically looks for this condition.  That new test in `builder_test.go` is IMHO _in addition_ to the very targeted unit test here confirming the `validateLink` method works as expected.   More the merrier!